test/DateYearDurationTest.cpp

```diff
@@ -0,0 +1,977 @@
+  EXPECT_EQ(duration0.toStringAndType().first, "PT0S");
+  EXPECT_STREQ(duration0.toStringAndType().second, XSD_DAYTIME_DURATION_TYPE);
+  duration0 = DayTimeDuration::parseXsdDayTimeDuration("PT0S");
+  EXPECT_EQ(duration0.toStringAndType().first, "PT0S");
+  EXPECT_STREQ(duration0.toStringAndType().second, XSD_DAYTIME_DURATION_TYPE);
+
+  // Test w.r.t. maximum values where we don't expect given the current bounds a
+  // normalization effect yet.
+  DayTimeDuration dMax =
+      DayTimeDuration::parseXsdDayTimeDuration("P1048575DT23H59M59.999S");
+  ASSERT_EQ(dMax.toStringAndType().first, "P1048575DT23H59M59.999S");
+  EXPECT_STREQ(dMax.toStringAndType().second, XSD_DAYTIME_DURATION_TYPE);
+  DayTimeDuration dMin =
+      DayTimeDuration::parseXsdDayTimeDuration("-P1048575DT23H59M59.999S");
+  ASSERT_EQ(dMin.toStringAndType().first, "-P1048575DT23H59M59.999S");
+  EXPECT_STREQ(dMin.toStringAndType().second, XSD_DAYTIME_DURATION_TYPE);
+
+  // provided invalid xsd:dayTimeDuration strings
+  ASSERT_THROW(DayTimeDuration::parseXsdDayTimeDuration("P0D0H0M0S"),
+               DurationParseException);
+  ASSERT_THROW(DayTimeDuration::parseXsdDayTimeDuration("0DT0H0M0S"),
+               DurationParseException);
+  ASSERT_THROW(DayTimeDuration::parseXsdDayTimeDuration("-P0D0HMS"),
+               DurationParseException);
+  ASSERT_THROW(DayTimeDuration::parseXsdDayTimeDuration("P0DABH0M0S"),
+               DurationParseException);
+}
+
+//______________________________________________________________________________
+TEST(DayTimeDuration, checkToAndFromBits) {
+  auto d1 = DayTimeDuration(DayTimeDuration::Type::Positive, 1, 23, 23, 59.99);
+  auto bits = d1.toBits();
+  d1 = DayTimeDuration::fromBits(bits);
+  DayTimeDuration::DurationValue dv = d1.getValues();
+  ASSERT_EQ(dv.days_, 1);
+  ASSERT_EQ(dv.hours_, 23);
+  ASSERT_EQ(dv.minutes_, 23);
+  EXPECT_NEAR(dv.seconds_, 59.99, 0.001);
+
+  auto d2 =
+      DayTimeDuration(DayTimeDuration::Type::Negative, 1048574, 3, 0, 0.99);
+  bits = d2.toBits();
+  d2 = DayTimeDuration::fromBits(bits);
+  dv = d2.getValues();
+  ASSERT_EQ(dv.days_, 1048574);
+  ASSERT_EQ(dv.hours_, 3);
+  ASSERT_EQ(dv.minutes_, 0);
+  EXPECT_NEAR(dv.seconds_, 0.99, 0.001);
+}
+
+//______________________________________________________________________________
+TEST(DayTimeDuration, DurationOverflowException) {
+  try {
+    [[maybe_unused]] auto d1 =
+        DayTimeDuration(DayTimeDuration::Type::Positive, 643917423, 4, 7, 1);
+    FAIL() << "DurationOverflowException was expected.";
+  } catch (const DurationOverflowException& e) {
+    EXPECT_STREQ(e.what(),
+                 "Overflow exception raised by DayTimeDuration, please provide "
+                 "smaller values for xsd:dayTimeDuration.");
+  }
+}
+
+//______________________________________________________________________________
+DayTimeDuration::DurationValue toAndFromMilliseconds(int days, int hours,
```

Comment:
What is the use case of this?

Reply:
W.r.t. large duration values, which are mostly internally normalized by conversion to milliseconds, it is quite complicated to test directly over the xsd:dayTimeduration strings, because the (input) duration string is different compared to the input string in terms of values. Thus i decided (while coding the `setValues()` and `getValues()`) to test this procedure explicitely.